src/Symfony/Component/Notifier/Bridge/MicrosoftTeams/MicrosoftTeamsOptions.php

```diff
@@ -0,0 +1,147 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\MicrosoftTeams;
+
+use Symfony\Component\Notifier\Bridge\MicrosoftTeams\Action\ActionInterface;
+use Symfony\Component\Notifier\Bridge\MicrosoftTeams\Section\Section;
+use Symfony\Component\Notifier\Bridge\MicrosoftTeams\Section\SectionInterface;
+use Symfony\Component\Notifier\Exception\InvalidArgumentException;
+use Symfony\Component\Notifier\Message\MessageOptionsInterface;
+use Symfony\Component\Notifier\Notification\Notification;
+
+/**
+ * @author Edouard Lescot <edouard.lescot@gmail.com>
+ * @author Oskar Stark <oskarstark@googlemail.com>
+ *
+ * @see https://docs.microsoft.com/en-us/outlook/actionable-messages/message-card-reference
+ */
+final class MicrosoftTeamsOptions implements MessageOptionsInterface
+{
+    private const MAX_ACTIONS = 4;
+
+    private $options = [];
+
+    public function __construct(array $options = [])
+    {
+        $this->options = $options;
+
+        $this->validateNumberOfActions();
+    }
+
+    public static function fromNotification(Notification $notification): self
+    {
+        $options = (new self())
+            ->title($notification->getSubject())
+            ->text($notification->getContent());
+
+        if ($exception = $notification->getExceptionAsString()) {
+            $options->section((new Section())->text($exception));
+        }
+
+        return $options;
+    }
+
+    public function toArray(): array
+    {
+        $options = $this->options;
+
+        // Send a text, not a message card
+        if (1 === \count($options) && isset($options['text'])) {
+            return $options;
+        }
+
+        $options['@type'] = 'MessageCard';
+        $options['@context'] = 'https://schema.org/extensions';
+
+        return $options;
+    }
+
+    public function getRecipientId(): ?string
+    {
+        return $this->options['recipient_id'] ?? null;
+    }
+
+    /**
+     * @param string $path The hook path (anything after https://outlook.office.com)
+     */
+    public function recipient(string $path): self
+    {
+        if (!preg_match('/^\/webhook\//', $path)) {
+            throw new InvalidArgumentException(sprintf('"%s" require recipient id format to be "/webhook/{uuid}@{uuid}/IncomingWebhook/{id}/{uuid}", "%s" given.', __CLASS__, $path));
+        }
+
+        $this->options['recipient_id'] = $path;
+
+        return $this;
+    }
+
+    /**
+     * @param string $summary Markdown string
+     */
+    public function summary(string $summary): self
+    {
+        $this->options['summary'] = $summary;
+
+        return $this;
+    }
+
+    public function title(string $title): self
+    {
+        $this->options['title'] = $title;
+
+        return $this;
+    }
+
+    public function text(string $text): self
+    {
+        $this->options['text'] = $text;
+
+        return $this;
+    }
+
+    public function themeColor(string $themeColor): self
+    {
```

Comment:
Verification of right heca color syntax migth be helpful here:

```php
if(!preg_match('/^#[0-9a-f]{3,6}$/i', $themeColor))
{
	throw new InvalidArgumentException(sprintf('MessageCard themeColor must have a valid hex color format.'));
}
```

Reply:
Implemented 👍